core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -723,14 +726,14 @@ class GroupCoordinator(val brokerId: Int,
         responseCallback(offsetMetadata.map { case (k, _) => k -> Errors.COORDINATOR_NOT_AVAILABLE })
       } else if (group.isStaticMemberFenced(memberId, groupInstanceId)) {
         responseCallback(offsetMetadata.map { case (k, _) => k -> Errors.FENCED_INSTANCE_ID })
-      } else if ((generationId < 0 && group.is(Empty)) || (producerId != NO_PRODUCER_ID)) {
+      } else if (generationId >= 0 && generationId != group.generationId) {
```

Comment:
The generation.id check needs to be put upfront before L734 as it treats txn commit same as assign mode consumer